trackintel/preprocessing/triplegs.py

```diff
@@ -200,124 +155,167 @@ def generate_trips(stps_input, tpls_input, gap_threshold=15, id_offset=0, print_
     spts_tpls.sort_values(by=['user_id', 'started_at'], inplace=True)
     spts_tpls['started_at_next'] = spts_tpls['started_at'].shift(-1)
     spts_tpls['activity_next'] = spts_tpls['activity'].shift(-1)
+    
+    if print_progress:
+        tqdm.pandas(desc='User trip generation')
+        trips = spts_tpls.groupby(['user_id'], 
+                                  group_keys=False, 
+                                  as_index=False).progress_apply(_generate_trips_user, gap_threshold=gap_threshold).reset_index(drop=True)
+    else:
+        trips = spts_tpls.groupby(['user_id'], 
+                                  group_keys=False, 
+                                  as_index=False).apply(_generate_trips_user, gap_threshold=gap_threshold).reset_index(drop=True)
+    trips['id'] = trips.index + id_offset
+    
+    # assign trip_id to tpls
+    ids = tpls.index
+    trip2tpl_map = trips[['id', 'tpls']].set_index('id').to_dict()['tpls']
+    ls = []
+    for key, values in trip2tpl_map.items():
+        for value in values:
+            ls.append([value, key])
+    temp = pd.DataFrame(ls, columns=['tplsIndex', 'trip_id'])
+    tpls = tpls.merge(temp, how='left', left_on='id', right_on='tplsIndex')
+    tpls.drop(columns='tplsIndex', inplace=True)
+    tpls['id'] = ids
+    tpls.set_index('id', inplace=True)
```

Comment:
I think the index of `tpls` is left untouched by the transformations above. If you can confirm, than we should delete these lines in order to not to change the index name of the user (I think at the moment we do not require that the index has the name 'id' but I am not sure...).

Reply:
solved. now using 'join' the indexes of tpls and spts are left untouched